### PR TITLE
[WIP] Drag to scroll

### PR DESCRIPTION
**Pull request by rbrtsmith**
Addressing #43 

* ~~A lingering bug whereby scroll position is lost when drag, stop then drag again.~~
* No tests for the time being.

### CODE REVIEW

File: src/components/Layout/index.jsx [hunk 1]
```diff
@@ -1,31 +1,65 @@
...
+    })
+  }
+
+  handleMouseUp() {
+    this.setState({ headerMouseDown: false })
+  }
+
+  handleHeaderMouseMove(e) {
+    if (this.state.headerMouseDown) {
+      const xDiff = this.state.dragStartX - e.clientX
```
Comment (WebSeed) [inline src/components/Layout/index.jsx]: `diffX`

File: src/components/Layout/Fixed.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import Sidebar from '../Sidebar'
+import Timeline from '../Timeline'
+import propTypes from './propTypes'
+
+const Layout = ({ isOpen, tracks, now, time, timebar, toggleTrackOpen, clickElement }) => (
```
Comment (WebSeed) [inline src/components/Layout/Fixed.jsx]: Better to call this something like `Default` if necessary as it isn't really "fixed" (in the CSS sense).

Also `src/components/Layout/Sticky.jsx` declares `class StickyLayout extends PureComponent` so this should probably be `class DefaultLayout extends PureComponent` for consistency.

File: src/components/Layout/index.jsx (hunk 1 above)
Comment (WebSeed) [inline src/components/Layout/index.jsx]: `deltaX`

File: src/components/Layout/index.jsx [hunk 3]
```diff
@@ -1,31 +1,65 @@
...
+  render() {
+    const drag = {
+      onDown: this.handleHeaderMouseDown,
+      onMove: this.handleHeaderMouseMove,
+      x: this.state.dragX,
+      mouseDown: this.state.headerMouseDown
+    }
+    return (
+      <div>
+        {React.cloneElement(this.props.children, { drag })}
```
Comment (WebSeed) [inline src/components/Layout/index.jsx]: ?

Comment (rbrtsmith) [inline src/components/Layout/index.jsx]: Allows us to pass props to children

**Comment by rbrtsmith** [comment]
I should add I'm not sure _how_ useful this is now I've been reminded by @atkinchris that combining shift with the mouse-wheel you can scroll horizontally.  What are your thoughts @WebSeed?

**Comment by atkinchris** [comment]
My comment about using _shift_ was an observation, not criticising the usefulness of this feature!

**Comment by WebSeed** [comment]
@atkinchris you are a monster, how dare you undermine this whole thing? But seriously, I had no idea about the mouse wheel plus shift thing and a daresay 95% of users will have either. If this is universal across browsers we could perhaps try and highlight the feature to them but generally I think drag to zoom is still a valid feature.

**Comment by atkinchris** [comment]
I'm going to close this as @rbrtsmith is no longer active on the library, and we don't have an internal need for the feature. The issue to add it (https://github.com/JSainsburyPLC/react-timelines/issues/43) remains, as does this code for reference.

**Pull request closed without merging**